src/main/java/com/minecolonies/coremod/client/gui/modules/ItemListModuleWindow.java

```diff
@@ -128,6 +136,28 @@ private void switchClicked(@NotNull final Button button)
         resourceList.refreshElementPanes();
     }
 
+    /**
+     * Fired when select or deselect has been clicked.
+     */
+    private void selectAll(@NotNull final boolean enable)
```

Comment:
If this method would be moved into the module the deselecting of all items could at least be optimized a lot (by using listsOfItems.clear() and sending a special message with the same effect).

Reply:
Still doesn't solve the problem, each item in the list will be sent to the request system causing performance issues with high numbers of items regardless of how the selecting and deselecting is handled.